File: examples/jte-example/src/main/java/de/tschuehly/example/jte/web/index/IndexViewComponent.java
package de.tschuehly.example.jte.web.index;

import de.tschuehly.spring.viewcomponent.core.component.ViewComponent;
import de.tschuehly.spring.viewcomponent.jte.ViewContext;

@ViewComponent
public class IndexViewComponent {
    public ViewContext render() {
        return new IndexView();
    }

    record IndexView() implements ViewContext {
    }
}

File: examples/kte-example/src/main/kotlin/de/tschuehly/kteviewcomponentexample/web/index/IndexViewComponent.kt
package de.tschuehly.kteviewcomponentexample.web.index

import de.tschuehly.spring.viewcomponent.core.component.ViewComponent
import de.tschuehly.spring.viewcomponent.kte.ViewContext

@ViewComponent
class IndexViewComponent {
    fun render() = IndexView()

    class IndexView : ViewContext
}